File: apps/php-sdk/src/Models/AgentTraceAgentIdentity.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentTraceAgentIdentity
{
    public function __construct(
        private readonly ?string $id = null,
        private readonly ?string $role = null,
        private readonly ?string $name = null,
        private readonly ?string $parentId = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'] ?? null,
            role: $data['role'] ?? null,
            name: $data['name'] ?? null,
            parentId: $data['parentId'] ?? null,
        );
    }

    public function getId(): ?string
    {
        return $this->id;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function getParentId(): ?string
    {
        return $this->parentId;
    }
}

File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentTraceAgentIdentity.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Identity of the agent that produced a trace event.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentTraceAgentIdentity {

    private String id;
    private String role;
    private String name;
    private String parentId;

    public String getId() { return id; }
    public String getRole() { return role; }
    public String getName() { return name; }
    public String getParentId() { return parentId; }

    @Override
    public String toString() {
        return "AgentTraceAgentIdentity{id=" + id + ", role=" + role + ", name=" + name + "}";
    }
}
